testjson/execution.go

```diff
@@ -456,9 +438,6 @@ func (p *Package) addTestEvent(event TestEvent) {
 		if tc.Test.IsSubTest() {
 			return
 		}
-
```

Comment:
No test has failed, so I see no reason to remove the output from memory here, even if the flag is off. If we do want to remove it if the flag is on, need to pass the cfg up to here.

Reply:
Ya, the reason for deleting the output here is that it could be very large, and holding all the output in memory could cause problems in CI.

I'm wondering if we can change the JUnit XML file handling to be more like the jsonfile handling. That's done here:

https://github.com/gotestyourself/gotestsum/blob/90a35cfb7751f7c6debf27f2fcf2fe07ba655176/cmd/handler.go#L39-L46

If we can write the output each time a test passes or fails then we only need to buffer output for running tests, instead of buffering all of the output for every run.

There will definitely be some challenges with that approach. The `handler.Event` runs after `Package.addTestEvent`, so the operation to "remove output" would need to move somewhere else, and there's currently no good place for it to move. The junit xml writer requires a lot more state than the jsonfile, so the implementation would be more involved.

Maybe the Junit XML writer could buffer the output itself, instead of relying on the buffer in `Package`.

Reply:
It might also be possible to change the interface between `handler` and `Package` so that the output is returned before it is removed, but I'm not sure yet what that would look like.